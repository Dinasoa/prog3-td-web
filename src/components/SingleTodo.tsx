import React from "react";

export const SingleTodo = () => {
    return(
        <>
        </>
    )
}