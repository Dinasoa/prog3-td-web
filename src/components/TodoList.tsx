import React from "react"
import { Todo } from "../model";
import '../style/style.css';

interface Props {
    todos: Todo[];
    setTodos: React.Dispatch<React.SetStateAction<Todo[]>>
    changeStatusToDone: (todo:Todo) => void;
    changeStatusTodo: (todo:Todo) => void;
}

export const TodoList: React.FC<Props> = ({ todos, setTodos , changeStatusToDone , changeStatusTodo}) => {
 
    return (
        <>
            <div className="todoList">
                <div className="todos" data-testid="to-do">
                <span className="title" >TO DO</span>
                    {todos.map((todo) =>  todo.isDone == false && (
                        <>
                        <li key={todo.id} data-testid="todo">
                            <input type="checkbox" data-testid = "input-checkbox" onChange={() => changeStatusToDone(todo)}/>
                            {todo.todo}
                        </li>                          
                        </>
                    ))}
                </div>
                <div className="todos">
                <span className="title">DONE</span>
                    {todos.map((todo ,index) => todo.isDone && (
                        <>
                        <li key={todo.id || index} data-testid='done'>
                            <input data-testid='input-done-checkbox' type="checkbox" onChange={() => changeStatusTodo(todo)} checked={true} />
                            {todo.todo}
                        </li>                          
                        </>
                    ))}
                </div>
            </div>
        </>
    )
}