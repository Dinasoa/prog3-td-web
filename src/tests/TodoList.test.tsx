import {fireEvent, render, screen} from "@testing-library/react";
import React from "react";
import {TodoList} from "../components/TodoList";
import {Todo} from "../model";
import renderer from "react-test-renderer";

it('Check if TodoList match the snapshot' , () => {
    const todos:Todo[] = [{id: 1 , todo: 'code', isDone:false}]
    const changeStatusTodo = jest.fn() ;
    const output = renderer.create(<TodoList todos={todos} setTodos={() => {}} changeStatusToDone={() => {}} changeStatusTodo={changeStatusTodo}/>) ;
    expect(output).toMatchSnapshot() ;
})

it('Check if the function changeStatusTodone should update the state to done' , () => {
   const changeStatus = jest.fn() ;
   const listOfTodo:Todo[] = [{id:1 , todo: 'this' , isDone:false}]
    const {getByTestId} = render(<TodoList todos={listOfTodo}
                                            setTodos={() => {
                                            }}
                                            changeStatusToDone={changeStatus}
                                            changeStatusTodo={() => {
                                            }}/> ) ;
   fireEvent.click(getByTestId('input-checkbox')) ;
   expect(changeStatus).toHaveBeenCalled() ;
   expect(listOfTodo.map((todo) => todo.isDone)).toBeTruthy() ;
})

it('Check if the function changeStatusTodo should update the state to to do ' , () => {
    const changeStatusToDone = jest.fn() ;
    const changeStatusTodo = jest.fn() ;
    const listOfTodo:Todo[] = [{id:1 , todo: 'this' , isDone:true}]
    const {getByTestId} = render(<TodoList todos={listOfTodo}
                                           setTodos={() => {
                                           }}
                                           changeStatusToDone={changeStatusToDone}
                                           changeStatusTodo={changeStatusTodo}/>) ;
    fireEvent.click(getByTestId('input-done-checkbox')) ;
    expect(changeStatusTodo).toHaveBeenCalled() ;
    expect(listOfTodo.map((todo) => todo.isDone == false)).toBeTruthy() ;
})

it('Check that todo task are displayed in the screen' , () => {
    const changeStatusToDone = jest.fn() ;
    const changeStatusTodo = jest.fn() ;
    const listOfTodo:Todo[] = [{id:1 , todo: 'this' , isDone:false}]
    const {getByTestId} = render(<TodoList todos={listOfTodo} setTodos={() => {}}
                                           changeStatusToDone={changeStatusToDone}
                                           changeStatusTodo={changeStatusTodo} />) ;
    const todo = getByTestId('todo') ;
    expect(todo).toBeInTheDocument() ;
})

it('Check that done task are displayed in the screen' , () => {
    const changeStatusToDone = jest.fn();
    const changeStatusTodo = jest.fn() ;
    const listOfDone:Todo[] = [{id:1 , todo: 'eat' , isDone:true}]
    const {getByTestId} = render(<TodoList todos={listOfDone} setTodos={() => {}} changeStatusToDone={changeStatusToDone}
                                           changeStatusTodo={changeStatusTodo}/>)
    const done = getByTestId('done') ;
    expect(done).toBeInTheDocument();
})

