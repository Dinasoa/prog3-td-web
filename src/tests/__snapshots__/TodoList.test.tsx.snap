// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if TodoList match the snapshot 1`] = `
<div
  className="todoList"
>
  <div
    className="todos"
    data-testid="to-do"
  >
    <span
      className="title"
    >
      TO DO
    </span>
    <li
      data-testid="todo"
    >
      <input
        data-testid="input-checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      code
    </li>
  </div>
  <div
    className="todos"
  >
    <span
      className="title"
    >
      DONE
    </span>
  </div>
</div>
`;
